crest/Assets/Crest/Crest/Scripts/Shapes/FFT/ShapeFFT.cs

```diff
@@ -397,12 +410,133 @@ public bool Validate(OceanRenderer ocean, ValidatedHelper.ShowMessage showMessag
+            message += $" Bake data resolution will be {target._resolution} x {target._resolution} x {scales}.";
+
+            message += "\n\n";
+            message += $"Period is {target._timeLoopLength}s.";
+            message += $" Frames per second setting is {target._timeResolution}.";
+            var frameCount = target._timeLoopLength * target._timeResolution;
+            message += $" Frame count is {target._timeLoopLength} x {target._timeResolution} = {frameCount}.";
+
+            message += "\n\n";
+            var pointsPerFrame = target._resolution * target._resolution * scales;
+            var channelCount = 4;
+            var bytesPerChannel = 4;
+            message += $"Total data size will be {pointsPerFrame * frameCount * channelCount * bytesPerChannel / 1048576f} MB.";
+
+            EditorGUILayout.HelpBox(message, MessageType.Info);
+        }
+
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+
+            var fft = target as ShapeFFT;
+
+            bool bakingEnabled = fft._enableBakedCollision;
+
+            if (bakingEnabled)
+            {
+                if (fft._spectrum == null)
+                {
+                    EditorGUILayout.HelpBox("A spectrum must be assigned to enable collision baking.", MessageType.Error);
+                    return;
+                }
+
+                BakeHelpBox(fft);
+            }
+
+            GUI.enabled = bakingEnabled;
+            OnInspectorGUIBaking();
+            GUI.enabled = true;
+        }
+
+        /// <summary>
+        /// Controls & GUI for baking.
+        /// </summary>
+        void OnInspectorGUIBaking()
+        {
+            if (OceanRenderer.Instance == null) return;
+
+            var bakeLabel = "Bake to asset";
+            var bakeAndAssignLabel = "Bake to asset and assign to current settings";
+            var selectCurrentSettingsLabel = "Select current settings";
+            if (OceanRenderer.Instance._simSettingsAnimatedWaves != null)
+            {
+                if (GUILayout.Button(bakeLabel))
+                {
+                    FFTBaker.BakeShapeFFT(target as ShapeFFT);
+                }
+
+                if (GUILayout.Button(bakeAndAssignLabel))
+                {
+                    var result = FFTBaker.BakeShapeFFT(target as ShapeFFT);
+                    if (result != null)
+                    {
+                        OceanRenderer.Instance._simSettingsAnimatedWaves.CollisionSource = SimSettingsAnimatedWaves.CollisionSources.BakedFFT;
+                        OceanRenderer.Instance._simSettingsAnimatedWaves._bakedFFTData = result;
+                        Selection.activeObject = OceanRenderer.Instance._simSettingsAnimatedWaves;
+
+                        // Rebuild ocean
+                        OceanRenderer.Instance.enabled = false;
+                        OceanRenderer.Instance.enabled = true;
```

Comment:
hm Instance is checked for null above. is it the reference to the settings that is throwing? that is unchecked and as a quick solution i've now disabled the "bake and assign" if there is no settings avail to assign to.

if it is actually this line and you have a repro, would you perhaps have a fix for it? just checking if a fix is obvious and can easily be made. if its not obvious let me konw if you have repro steps and ill try here.

Reply:
The line before it disables the ocean and which Instance is set to null. We can either add a method on the ocean to handle this since it then won't rely on the static reference, or we could use hashy.

Not sure how to add to this PR myself, but here is the method on OR:

```csharp
        public void Rebuild()
        {
            enabled = false;
            enabled = true;
        }
```

Reply:
Oh I did think about that, but it never happened for me so I dismissed it. Weird. That makes sense though. I'll call that function instead.

Reply:
added that function to OR, thanks

re adding to pr, in retrospect i should probably create a branch in this depot for us to iterate for cases like this. to push to the branch in this pr, you'd need to ask for write access to the other remote and add the remote to your local git workspace. not sure its worth changing anything now but good to note going forwards.

Reply:
Thanks. Looks good.